modules/plugin/geotiff/src/main/java/org/geotools/gce/geotiff/GeoTiffReader.java

```diff
@@ -968,7 +968,7 @@ protected final GridCoverage2D createCoverage(
 
         Category noDataCategory = null;
         final Map<String, Object> properties = new HashMap<>();
-        if (!Double.isNaN(noData)) {
+        if (Double.isFinite(noData)) {
```

Comment:
I am not sure that this is a 100% valid case, but it makes sense, since it looks like Range can't contain NaNs and Infinite values.